Singularity/Singularity/Screen/Button.cs

```diff
@@ -0,0 +1,219 @@
+        private readonly string mButtonText;
+        private readonly Vector2 mPosition;
+        private readonly int mWidth;
+        private readonly int mHeight;
+        private readonly SpriteFont mFont;
+
+        // distinguish between mouse over hover or not
+        private Color mColor;
+
+        private Rectangle mBounds;
+        private bool mClicked;
+
+        // events that can be sent out by buttons
+        // an event handler must be created in the class that 
+        // has an instance of buttons is created in
+        // order to receive these events
+        public event EventHandler ButtonReleased;
+        public event EventHandler ButtonHovering;
+        public event EventHandler ButtonClicked;
+
+
+        /// <summary>
+        /// Creates a button using a Texture2D
+        /// </summary>
+        /// <param name="scale"> scale of the texture</param>
+        /// <param name="buttonTexture"></param>
+        /// <param name="position"></param>
+        public Button(float scale, Texture2D buttonTexture, Vector2 position)
+        {
+            mIsText = false;
+            mScale = scale;
+            mButtonTexture = buttonTexture;
+            mPosition = position;
+            mWidth = mButtonTexture.Width;
+            mHeight = mButtonTexture.Height;
+            mColor = Color.White;
+            CreateRectangularBounds();
+        }
+
+        /// <summary>
+        /// Creates a Button made of text
+        /// </summary>
+        /// <param name="buttonText">text that button will appear as</param>
+        /// <param name="font"></param>
+        /// <param name="position"></param>
+        public Button(string buttonText, SpriteFont font, Vector2 position)
+        {
+            mIsText = true;
+            mButtonText = buttonText;
+            mFont = font;
+            mPosition = position;
+            mWidth = (int)mFont.MeasureString(mButtonText).X;
+            mHeight = (int)mFont.MeasureString(mButtonText).Y;
+            mColor = Color.White;
+            CreateRectangularBounds();
+        }
+
+
+        /// <summary>
+        /// Creates the bounding box that the button is contained in
+        /// </summary>
+        private void CreateRectangularBounds()
+        {
+            mBounds = new Rectangle((int)mPosition.X, (int)mPosition.Y, mWidth, mHeight);
+        }
+
+        /// <summary>
+        /// Sends out event that button has been left click released
+        /// </summary>
+        protected virtual void OnButtonReleased()
+        {
+            if (ButtonReleased != null)
+            {
+                ButtonReleased(this, EventArgs.Empty);
+            }
+
+        }
+
+
+        /// <summary>
+        /// Sends out event that mouse is hovering over the button
+        /// </summary>
+        protected virtual void OnButtonHovering()
+        {
+            if (ButtonHovering != null)
+            {
+                ButtonHovering(this, EventArgs.Empty);
```

Comment:
how do those events get noticed from the listener if they're only being created?
(highly possible that I'm just not aware of how events are being handled in c# specifically)

Reply:
id ask anyone with questions to contact me or look online since documentation on how to use this can take up quite some space

Reply:
fair enough.

Reply:
okay, has a short step by step now of how to use. with any further questions contact me :)